### PR TITLE
[ci] Actually build all software in the sw_build job

### PR DESCRIPTION
**Pull request by a-will**
The sw_build job was missing many targets, as the build and test commands resolve wildcards very differently, even with --nobuild_tests_only.

### CODE REVIEW

File: azure-pipelines.yml
Comment (a-will) [inline azure-pipelines.yml:272]: This took some experimentation... To make the correct query to get the ELF output file, we have to start from a node that is built for our usual target platform (the host) and follow the edges in the graph across the configuration transition (to riscv32).

**Review by rswarbrick — APPROVED** [review]

File: azure-pipelines.yml
Comment (marnovandermaas) [inline azure-pipelines.yml:271]: Nit: can we put this sentence on the same line?

File: azure-pipelines.yml
Comment (marnovandermaas) [reply azure-pipelines.yml:272]: > To make the correct query to get the ELF output file, we have to start from a node that is built for our usual target platform (the host) and follow the edges in the graph across the configuration transition (to riscv32).

Maybe we can add this explanation as a comment in the code?

File: azure-pipelines.yml
Comment (a-will) [reply azure-pipelines.yml:271]: Er... do we not attempt to do line limits in these files for readability? I know I didn't try very hard with some of the bash lines, but...

Comment (marnovandermaas) [reply azure-pipelines.yml:271]: I guess it's ok, just looks weird to me when there are other lines that are not wrapped. CI does not seem to mind either way, so I guess this need not be blocking.

File: azure-pipelines.yml
Comment (a-will) [reply azure-pipelines.yml]: Done.

**Review by marnovandermaas — APPROVED** [review]
Thanks for this fix! Let's wait for CI to pass before merging.

**Comment by alphan** [comment]
Thanks @a-will !

**Pull request merged**
